eventsources/sources/file/start.go

```diff
@@ -138,6 +158,32 @@ func (el *EventListener) listenEvents(ctx context.Context, dispatch func([]byte,
 				// watcher stopped watching file events
 				return fmt.Errorf("fs watcher stopped for %s", el.GetEventName())
 			}
+			if (fileEventSource.WatchPathConfig.Recursive) && (event.Has(fsnotify.Create)) {
+				fileinfo, err := os.Stat(event.Name)
+				if err != nil {
+					log.Errorw("failed getting filestat", zap.Error(err))
+				}
+				if fileinfo.IsDir() {
+					err = watcher.Add(event.Name)
```

Comment:
1. If there is a dir create with sub directories (folder A contains B, B contains C), how many events are supposed to get? will it have duplicated processing?
2. No eviction mechanism for deleting?